lib/client.js

```diff
@@ -99,7 +99,16 @@ Client.prototype._request = function (method, url, opts, cb) {
   });
 
   request[method](url, opts, function (err, res, body) {
-    var statsName;
+    var statsName,
+        cacheableRes = {};
+
+    if (res) {
```

Comment:
If we move this after the current early return for the error then there's no need to check for a `res`.